brage-import/src/main/java/no/sikt/nva/brage/migration/lambda/BrageEntryEventConsumer.java

```diff
@@ -91,15 +102,63 @@ private static Publication getOnlyElement(List<Publication> publications) {
                        fail.getException()));
     }
 
-    private boolean publicationWithCristinIdentifierAlreadyExists(Publication publication) {
+    private boolean shouldMergePublications(Publication publication) throws NotFoundException {
         var cristinIdentifier = getCristinIdentifier(publication);
         if (nonNull(cristinIdentifier)) {
-            publicationsByCristinIdentifier = getPublicationsByCristinIdentifier(cristinIdentifier);
-            return !publicationsByCristinIdentifier.isEmpty();
+            publicationsToMerge = getPublicationsByCristinIdentifier(cristinIdentifier);
+            return !publicationsToMerge.isEmpty();
+        }
+
+        return existingPublicationHasSameDoi(publication);
+    }
+
+    private boolean existingPublicationHasSameDoi(Publication publication)
+        throws NotFoundException {
+        var doi = publication.getEntityDescription().getReference().getDoi();
+
+        if (nonNull(doi)) {
+            var publications = searchForPublicationsByDoi(doi);
+            if (publications.isEmpty()) {
+                return false;
+            }
+
+            // Fetch first publication regardless of how many hits
+            var publicationByDoi = publications.get(0);
```

Comment:
Not a huge fan of comments to explain code. They start telling lies quickly, and then linger in the code and creates confusion. I really agree with this: https://youtu.be/2a_ytyt9sf8?si=ku7A4QCXjEekOyTZ

Either write a test that specifies that we are ignoring everything except the first hit. OR rename this variable to indicate that we are ignoring everything except first hit.